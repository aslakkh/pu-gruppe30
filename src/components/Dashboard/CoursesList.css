.CoursesList{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.ListElementHere{
    display: flex;
    flex-direction: row;
}
