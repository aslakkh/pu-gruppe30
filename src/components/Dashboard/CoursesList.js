import React, { Component } from 'react'
import {ListGroup, ListGroupItem, Button} from 'react-bootstrap'
import {disableCourse} from '../../helpers/auth'
import {styles} from './CoursesListStyles.js'


/*      Component for displaying a users courses   

Returns courses as ListGroup
Renders upon change in DB

*/ 


export default class CoursesList extends Component {

	constructor(props) {
		super(props);
		this.state={
			courses: this.props.courses
		}
	}


	componentWillReceiveProps(nextProps){
		this.setState({
				courses: nextProps.courses,
			});
		
}



	handleClick(key){
		disableCourse(key); //sets "active" to false in database
	}



    //Component renders a header, followed by the keys of json object retrieved from firebase mapped to list
	render(){
		if(this.state.courses === undefined || this.state.courses === null){
			return(
				<h5>You have no active courses. Add a course above.</h5>
			);
		}
		else{
			return(
				<ListGroup>
						{Object.keys(this.state.courses).map((key) => {
							return (
								<ListGroupItem key={key} style={styles.coursesList}> 
									<div style={styles.listElement}>{key}</div>
									<Button bsStyle="danger" onClick={() => this.handleClick(key)}>Delete</Button>
								</ListGroupItem>
							);
								
						})} 
				</ListGroup>
		);
		}
	}

}


