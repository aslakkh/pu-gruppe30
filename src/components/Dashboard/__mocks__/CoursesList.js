import React, { Component } from 'react'
import {ListGroup, ListGroupItem, Button} from 'react-bootstrap'
import {disableCourse, courseExistsAtRoot} from '../../helpers/auth'
import {CourseWarning} from '../various/CourseWarning'
import {styles} from './CoursesListStyles.js'


/*      Component for displaying a users courses   

Returns courses as ListGroup
Renders upon change in DB

TODO: 
- Ask for confirmation on Delete
- Submit with enter
- Add courses in chronological order?
- ()
Could probably be split into a Course-component for each courses
*/ 

//http://stackoverflow.com/questions/40494787/get-user-object-properties-from-firebase-in-react

export default class CoursesList extends Component {

	constructor(props) {
		super(props);
		this.state={
			
			courses: this.props.courses
		}
	}

	componentDidMount(){
		//should be implemented in auth.js
		/*var self = this;
		var userUid = getUserUid(); //gets userid of user currently logged in

        //following code gets users courses from firebase and updates the components state
        //Should maybe be implemented in a function in auth.js
		self.firebaseRef = firebase.database().ref().child("users/"+userUid+"/courses");
		self.firebaseRef.orderByChild("active").equalTo(true).on('value', snap => {
			console.log(snap.val());
			self.setState({
				
				courses: snap.val() //sets courses to whatever is at users/userUid/courses in database
			});
		});*/
		
	}

	componentWillReceiveProps(nextProps){
		this.setState({
			courses: nextProps.courses
		});

}

	componentWillUnmount(){

		//this.firebaseRef.off('value');
	}

	//handles click
	handleClick(key){
		return true;
	}



    //Component renders a header, followed by the keys of json object retrieved from firebase mapped to list
	render(){
		
		if(this.state.courses === undefined || this.state.courses === null){
			return(
				<h5>You have no active courses. Add a course below.</h5>
			);
		}
		else{
			return(
				<ListGroup>
						{Object.keys(this.state.courses).map((key) => {
							
							return <ListGroupItem key={key} /*className="CoursesList"*/ style={styles.coursesList}> 
								<div style={styles.listElement}>{key}</div>
								<Button bsStyle="danger" onClick={() => this.handleClick(key)}>Delete</Button>
								</ListGroupItem>
			
							
						})} 
				</ListGroup>
		);
		}
	}

}


