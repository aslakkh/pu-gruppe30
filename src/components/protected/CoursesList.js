import React, { Component } from 'react'
import {ListGroup, ListGroupItem, Button} from 'react-bootstrap'
import { getUserUid, removeCourse} from '../../helpers/auth'
import firebase from 'firebase'
import './CoursesList.css'


/*      Component for displaying a users courses   

Returns courses as ListGroup
Renders upon change in DB

TODO: 
FIX ERROR REMOVING LAST ELEMENT IN LIST / LOADING A USER WITH NO COURSES
FIX WARNING MESSAGE ON LOAD AFTER VIEWING COURSES
Could probably be split into a Course-component for each courses
*/ 

//http://stackoverflow.com/questions/40494787/get-user-object-properties-from-firebase-in-react



export default class CoursesList extends Component {

	constructor(props) {
		super(props);
		this.state={
			courses: []
		}
	}

	componentDidMount(){
		//should be implemented in auth.js
		var self = this;
		var userUid = getUserUid(); //gets userid of user currently logged in

        //following code gets users courses from firebase and updates the components state
        //Should maybe be implemented in a function in auth.js
		const coursesRef = firebase.database().ref().child("users/"+userUid+"/courses");
		coursesRef.on('value', snap => {
			self.setState({
				courses: snap.val() //sets courses to whatever is at users/userUid/courses in database
			});
		});
	}

	//handles click
	handleClick(key){
		removeCourse(key);
	}



    //Component renders a header, followed by the keys of json object retrieved from firebase mapped to list
	render(){
		

		
		return(
			<div>
					
					<ListGroup >
							{Object.keys(this.state.courses).map((key) => {
								return <ListGroupItem key={key} className="CoursesList"> 
										{key} 
										<Button bsStyle="danger" onClick={() => this.handleClick(key)}>Delete</Button>
										</ListGroupItem>;
							})} 
					</ListGroup>
			</div>
		);
		
	}

}


