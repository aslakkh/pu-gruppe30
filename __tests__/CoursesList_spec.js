import React from 'react';
import renderer from 'react-test-renderer';
import CoursesList from '../src/components/Dashboard/CoursesList.js';

const courses = {
    IT5555: {
        active: true,
        goal: 0,
        goals: {},
        oldGoals: {},
        time: 0,
    },
    TDT0000: {
        active: true,
        goal: 0,
        goals: {},
        oldGoals: {},
        time: 0,
    }
};

test('CoursesList renders ListGroup', () => {
    const component = renderer.create(
        <CoursesList courses={courses} />
    ).toJSON();
    expect(component).toMatchSnapshot();
});