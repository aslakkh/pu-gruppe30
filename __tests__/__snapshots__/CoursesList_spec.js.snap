// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesList renders ListGroup 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      IT5555
    </div>
    <button
      className="btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </li>
  <li
    className="list-group-item"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      TDT0000
    </div>
    <button
      className="btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </li>
</ul>
`;
